api/server/router/container/container_routes.go

```diff
@@ -693,15 +693,22 @@ func (s *containerRouter) wsContainersAttach(ctx context.Context, w http.Respons
 		return conn, conn, conn, nil
 	}
 
+	useStdin, useStdout, useStderr := true, true, true
+	if versions.GreaterThanOrEqualTo(version, "1.42") {
+		useStdin = httputils.BoolValue(r, "stdin")
+		useStdout = httputils.BoolValue(r, "stdout")
+		useStderr = httputils.BoolValue(r, "stderr")
+	}
```

Comment:
Trying to think if there's a way to make the change less disruptive, but not sure how to;

- API < 1.41 enabled these by default (which didn't match the docs (see my other comment)), so I'd consider that to be a bug
- But has been around for a long time, so we probably shouldn't change for existing API versions

So in the old API versions; if I didn't look at the API docs, and forgot to set `?stdin=1&stdout=1&stderr=1`, then I would _still_ have all three attached.

In the new API version; I _MUST_ set all three to preserve the old behavior; which matches what was documented, but (see above) "just worked" before.


So, yeah, I don't see a clear way to migrate between one and the other, other than calling it out explicitly in the release notes 🤔 🤷‍♂️